rtl/eclrtl/eclrtl.cpp

```diff
@@ -4851,12 +4851,11 @@ class CStrRegExprFindInstance : implements IStrRegExprFindInstance
         {
             if (_keep)
             {
-                sample = (char *)rtlMalloc(_len + 1);  //required for findstr
+                sample = (char *)rtlMalloc(_len + 1);
                 memcpy(sample, _str + _from, _len);
-                sample[_len] = (char)NULL;
```

Comment:
I've added that line back in.

I see another problem here.. If from is >0, then it copies in a shorter string but terminates at the position of the original string end.